sdk/storage/Azure.Storage.DataMovement/src/BlobDirectoryTransferJob.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.DataMovement.Models;
+
+namespace Azure.Storage.DataMovement
+{
+    // Consideration: Should we make a Job type for each type of job?  We might have some overlap
+    // so we would have BlobSingleUploadTransferJob, BlobSingleDownloadTransferJob, BlobSIngleCopyTransferJob,
+    // BlobUploadDirectoryTransferJob
+    internal class BlobDirectoryTransferJob : StorageTransferJob
+    {
+        private string _localPath;
+
+        // if the upload path is local or will be the destination of the download
+        public string localPath => _localPath;
+
+        internal BlobDirectoryClient _destinationBlobClient;
+
+        public BlobDirectoryClient destinationBlobClient => _destinationBlobClient;
+
+        private BlobDirectoryClient _sourceBlobClient;
+
+        public BlobDirectoryClient sourceBlobClient => _sourceBlobClient;
+
+        internal BlobUploadOptions _uploadOptions;
+
+        public readonly CopyMethod _copyMethod;
+
+        // Should only be used for upload options. Felt redudant
+        // to create a whole other class that inherited this just
+        // for uploads. Is it worth it to make a whole other class
+        // for each operation type.
+        public BlobUploadOptions options => _uploadOptions;
```

Comment:
coupling is more dangerous than duplication. duplication can be handled by super types or interfaces to some extent. Coupling is hard to undo.

Reply:
done. I made a transfer job for each type.